src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  const newState = { ...state };
```

Comment:
The `newState` object should be recreated inside the loop for each action to avoid mutating the same object over the iterations. Otherwise, all entries in `stateHistory` will reference the same object, which is not the intended behavior.